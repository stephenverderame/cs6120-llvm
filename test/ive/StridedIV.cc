// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    for (int i = 0; i < 5000; i += m) {
        int p = m + i * 10;
        for (int j = 0; j < i; j += p) {
            int k = j + i * p;
            printf("%d\n", k);
        }
    }
}
// NOLINTEND