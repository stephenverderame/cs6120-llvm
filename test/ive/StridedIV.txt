4400
16800
37200
65600
102000
146400
198800
259200
327600
404000
488400
580800
681200
789600
906000
1030400
1162800
1303200
1451600
1608000
1772400
1944800
2125200
2313600
2510000
2714400
2926800
3147200
3375600
3612000
3856400
4108800
4369200
4637600
4914000
5198400
5490800
5791200
6099600
6416000
6740400
7072800
7413200
7761600
8118000
8482400
8854800
9235200
9623600
10020000
10424400
10836800
11257200
11685600
12122000
12566400
13018800
13479200
13947600
14424000
14908400
15400800
15901200
16409600
16926000
17450400
17982800
18523200
19071600
19628000
20192400
20764800
21345200
21933600
22530000
23134400
23746800
24367200
24995600
25632000
26276400
26928800
27589200
28257600
28934000
29618400
30310800
31011200
31719600
32436000
33160400
33892800
34633200
35381600
36138000
36902400
37674800
38455200
39243600
40040000
40844400
41656800
42477200
43305600
44142000
44986400
45838800
46699200
47567600
48444000
49328400
50220800
51121200
52029600
52946000
53870400
54802800
55743200
56691600
57648000
58612400
59584800
60565200
61553600
62550000
63554400
64566800
65587200
66615600
67652000
68696400
69748800
70809200
71877600
72954000
74038400
75130800
76231200
77339600
78456000
79580400
80712800
81853200
83001600
84158000
85322400
86494800
87675200
88863600
90060000
91264400
92476800
93697200
94925600
96162000
97406400
98658800
99919200
101187600
102464000
103748400
105040800
106341200
107649600
108966000
110290400
111622800
112963200
114311600
115668000
117032400
118404800
119785200
121173600
122570000
123974400
125386800
126807200
128235600
129672000
131116400
132568800
134029200
135497600
136974000
138458400
139950800
141451200
142959600
144476000
146000400
147532800
149073200
150621600
152178000
153742400
155314800
156895200
158483600
160080000
161684400
163296800
164917200
166545600
168182000
169826400
171478800
173139200
174807600
176484000
178168400
179860800
181561200
183269600
184986000
186710400
188442800
190183200
191931600
193688000
195452400
197224800
199005200
200793600
202590000
204394400
206206800
208027200
209855600
211692000
213536400
215388800
217249200
219117600
220994000
222878400
224770800
226671200
228579600
230496000
232420400
234352800
236293200
238241600
240198000
242162400
244134800
246115200
248103600
