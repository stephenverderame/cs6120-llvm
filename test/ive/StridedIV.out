Basic IV:   %13 = phi i32 [ 0, %7 ], [ %16, %12 ]
Basic IV:   %14 = add nuw nsw i32 %13, %8
Basic IV:   %16 = add nuw nsw i32 %13, %5
Basic IV:   %10 = add nuw nsw i32 %3, 20
Basic IV:   %3 = phi i32 [ 0, %0 ], [ %10, %9 ]
Derived IV: %4 = %3 * 10
Derived IV: %5 = %3 * 10 + 20
